server.go

```diff
@@ -3758,7 +3762,6 @@ func (s *server) peerTerminationWatcher(p *peer.Brontide, ready chan struct{}) {
 // peer.
 func (s *server) connectToPersistentPeer(pubKeyStr string) {
 	s.mu.Lock()
-	defer s.mu.Unlock()
```

Comment:
Why no longer defer here?

Reply:
added back 👍  (initially didn't have the reconnection part at the end in a goroutine which is why i moved this)